contracts/EpochTokenLocker.sol

```diff
@@ -121,12 +133,17 @@ contract EpochTokenLocker {
         return balance;
     }
 
+    function hasValidWithdrawRequest(address user, address token) public view returns(bool) {
+        return balanceStates[user][token].pendingWithdraws.stateIndex < getCurrentBatchId() &&
+            balanceStates[user][token].pendingWithdraws.stateIndex > 0;
+    }
+
     /**
      * internal functions
      */
-    function addBalanceAndPostponeWithdraw(address user, address token, uint amount) internal {
-        if (balanceStates[user][token].pendingWithdraws.stateIndex < getCurrentBatchId()) {
-            balanceStates[user][token].pendingWithdraws.stateIndex = getCurrentBatchId();
+    function addBalanceAndBlockWithdrawForThisBatch(address user, address token, uint amount) internal {
```

Comment:
Do we always want to do the two things together, or is it better to have two different methods? The good think about one method is that you cannot "forget" to block the withdraw when you change the balance.

Reply:
Right, I also like the point you mentioned!